### ISSUE TITLE
TypeError: Blob is not a constructor

### ISSUE THREAD

**Issue by Alain00** [devongovett/blob-stream#5]
I'm getting the following error

```
\blob-stream\index.js:33
    this._blob = new Blob(this._chunks, {
                 ^

TypeError: Blob is not a constructor
    at BlobStream.toBlob (H:\Work\projects\node\subzero\node_modules\blob-stream\index.js:33:18)
    at BlobStream.toBlobURL (H:\Work\projects\node\subzero\node_modules\blob-stream\index.js:48:35)
    at BlobStream.stream.on (H:\Work\projects\node\subzero\routes\cpanel\orderadmin.router.js:185:32)
    at BlobStream.emit (events.js:203:15)
    at finishMaybe (_stream_writable.js:646:14)
    at afterWrite (_stream_writable.js:486:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

This is the code

```js
const PDFDocument = require('pdfkit');
const blobStream  = require('blob-stream');
let doc = new PDFDocument();
const stream = doc.pipe(blobStream());
doc.text("HOLA MUNDO!!");
doc.end();
stream.on('finish', ()=>{
    const url = stream.toBlobURL('application/pdf');
    res.send(url);
})
```

**Comment by dhatawesomedude** [comment]
@Alain00 I'm facing this same issue. Were you able to figure this out ?

**Comment by eveningkid** [comment]
Hey guys,

**tldr: Blob is web-only**

I had the same issue and it just occurred to me that `Blob` does not exist on Node side (it's using buffers instead).

Basically, it is already a stream so you don't need to make a blob out of it.

In my case, I am creating a PDF from Express and I just had to pipe `res` to the document instance:

```javascript
router.get('/document.pdf', async (req, res) => {
  const doc = new PDFDocument();
  
  // vv The following line is the one you're looking for
  doc.pipe(res);

  doc
    .text('And here is some wrapped text...', 100, 300)
    .font('Times-Roman', 13);

  doc.end();

  res.writeHead(200, {
    'Content-Type': 'application/pdf',
  });
});
```

If this can be of any help :)

**Comment by Ryan-F3107** [comment]
facing the same issue too, I wanted to create a pdf on backend and then use the blob and save it on firebase storage.

**Comment by dansalado** [comment]
Hi! Maybe these could help :)

https://github.com/foliojs/pdfkit/issues/625

https://stackoverflow.com/questions/35658438/node-js-piping-pdfkit-to-a-memory-stream

**Comment by adeayobandung** [comment]
> Hey guys,
> 
> **tldr: Blob is web-only**
> 
> I had the same issue and it just occurred to me that `Blob` does not exist on Node side (it's using buffers instead).
> 
> Basically, it is already a stream so you don't need to make a blob out of it.
> 
> In my case, I am creating a PDF from Express and I just had to pipe `res` to the document instance:
> 
> ```js
> router.get('/document.pdf', async (req, res) => {
>   const doc = new PDFDocument();
>   
>   // vv The following line is the one you're looking for
>   doc.pipe(res);
> 
>   doc
>     .text('And here is some wrapped text...', 100, 300)
>     .font('Times-Roman', 13);
> 
>   doc.end();
> 
>   res.writeHead(200, {
>     'Content-Type': 'application/pdf',
>   });
> });
> ```
> 
> If this can be of any help :)

Many Thanks, this is help me more.

**Comment by Ian-Balijawa** [comment]
blobs are web only

use something like this incase you're in an express/nodejs server

```
import PDFDocument from "pdfkit";
import express from "express";
const router = express.Router();

router.get( '/', async ( req, res ) => {
	const doc = new PDFDocument();

	doc.pipe( res );

	doc.text( 'And here is some wrapped text...', 100, 300 ).font( 'Times-Roman', 13 );

	doc.end();

	res.writeHead( 200, {
		'Content-Type': 'application/pdf',
	} );
} );

export default router;
```

**Comment by Lukenickerson** [comment]
Looks like `Blob` was added to node in v14.18.0 https://nodejs.org/api/all.html#all_buffer_class-blob
